src/teleop/gui/src/components/ToggleButton.vue

```diff
@@ -0,0 +1,116 @@
+<template>
+  <label
+    :for="id + '_button'"
+    :class="{ active: currentState }"
+    class="toggle__button"
+  >
+    <span v-if="currentState" class="toggle__label">{{ labelEnableText }}</span>
+    <span v-if="!currentState" class="toggle__label">{{
+      labelDisableText
+    }}</span>
+
+    <input :id="id + '_button'" v-model="checkedValue" type="checkbox" />
+    <span class="toggle__switch"></span>
+  </label>
+</template>
+
+<script>
+export default {
+  props: {
+    labelEnableText: {
+      type: String,
+      required: true,
+    },
+    labelDisableText: {
+      type: String,
+      required: true,
+    },
+    id: {
+      type: String,
+      default: "primary",
+    },
+    currentState: {
+      type: Boolean,
+      required: true,
+    },
+  },
+  computed: {
+    checkedValue: {
+      get() {
+        return this.currentState;
+      },
+      set(newValue) {
+        this.$emit("change", newValue);
```

Comment:
This is a nit for sure, but ESLint enforces strictly defining the emits of Vue components, so this is something that we should probably begin implementing in our code.
https://vuejs.org/guide/components/events.html#declaring-emitted-events